modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java

```diff
@@ -0,0 +1,359 @@
+        typeMapping.put("enum", "NSString");
+        typeMapping.put("array", "List");
+        typeMapping.put("set", "Set");
+        typeMapping.put("boolean", "Boolean");
+        typeMapping.put("string", "String");
+        typeMapping.put("int", "Int");
+        typeMapping.put("long", "Long");
+        typeMapping.put("float", "Float");
+        typeMapping.put("byte", "Byte");
+        typeMapping.put("short", "Short");
+        typeMapping.put("char", "Char");
+        typeMapping.put("double", "Double");
+        typeMapping.put("object", "Any");
+        typeMapping.put("file", "File");
+        typeMapping.put("binary", "String");
+        typeMapping.put("ByteArray", "String");
+        typeMapping.put("date-time", "Date");
+        typeMapping.put("DateTime", "Date");
+
+        instantiationTypes.put("array", "ListBuffer");
+        instantiationTypes.put("map", "HashMap");
+
+        setReservedWordsLowerCase(
+                Arrays.asList(
+                        // local variable names used in API methods (endpoints)
+                        "path", "contentTypes", "contentType", "queryParams", "headerParams",
+                        "formParams", "postBody", "mp", "basePath", "apiInvoker",
+
+                        // scala reserved words
+                        "abstract", "case", "catch", "class", "def", "do", "else", "extends",
+                        "false", "final", "finally", "for", "forSome", "if", "implicit",
+                        "import", "lazy", "match", "new", "null", "object", "override", "package",
+                        "private", "protected", "return", "sealed", "super", "this", "throw",
+                        "trait", "try", "true", "type", "val", "var", "while", "with", "yield")
+        );
+    }
+
+    /**
+     * Gatling does not need the models to have escaped words as it builds models dynamically instead of through
+     * an instance of the object.
+     *
+     * @return the escaped term
+     */
+    @Override
+    public String escapeReservedWord(String name) {
+        return name;  // add an underscore to the name
+    }
+
+    /**
+     * Location to write model files.  You can use the modelPackage() as defined when the class is
+     * instantiated
+     */
+    public String modelFileFolder() {
+        return outputFolder + "/" + sourceFolder + "/" + modelPackage().replace('.', File.separatorChar);
+    }
+
+    /**
+     * Location to write api files.  You can use the apiPackage() as defined when the class is
+     * instantiated
+     */
+    @Override
+    public String apiFileFolder() {
+        return outputFolder + "/" + sourceFolder + "/" + apiPackage().replace('.', File.separatorChar);
+    }
+
+    /**
+     * Modifies the swagger doc to make mustache easier to use
+     *
+     * @param swagger
+     */
+    @Override
+    public void preprocessSwagger(Swagger swagger) {
+        if (swagger == null || swagger.getPaths() == null) {
+            return;
```

Comment:
Does this throw elsewhere?

Reply:
i copied this from another template once i started.  I can remove this now.